modules/root/root_v2_api.rb

```diff
@@ -15,8 +15,9 @@ class Proxy::RootV2Api < Sinatra::Base
       attributes = [:http_enabled, :https_enabled, :settings, :state]
 
       plugins = enabled_plugins.inject({}) do |hash, plugin|
-        result = Hash[attributes.map { |attribute| [attribute, plugin[attribute]] }]
-        result[:capabilities] = process_capabilities(plugin[:state], plugin[:capabilities])
+        result = Hash[attributes.map { |attribute| [attribute, plugin[attribute]] }.delete_if { |k, v| v.nil? || (v.instance_of?(Hash) && v.empty?) }]
+        capabilities = process_capabilities(plugin[:state], plugin[:capabilities])
+        result[:capabilities] = capabilities unless capabilities.empty?
```

Comment:
IMHO we should always have a list, even if it's empty. Application code is much easier that way.

Reply:
You can't assume that a string coming from (bad) internet has some empty elements. When parsing, be defensive. While generally in programming it's good practice to avoid null values, I do think in this case it's considered harmful.